src/client.rs

```diff
@@ -71,81 +22,28 @@ pub(crate) fn check_key_len(key: &str) -> Result<(), MemcacheError> {
 }
 
 impl Client {
-    #[deprecated(since = "0.10.0", note = "please use `connect` instead")]
-    pub fn new<C: Connectable>(target: C) -> Result<Self, MemcacheError> {
-        return Self::connect(target);
+    pub fn with_pool(pool: Pool<ConnectionManager>) -> Self {
+        Self(pool)
     }
 
-    pub fn with_pool_size<C: Connectable>(target: C, size: u32) -> Result<Self, MemcacheError> {
-        let urls = target.get_urls();
-        let mut connections = vec![];
-        for url in urls {
-            let parsed = Url::parse(url.as_str())?;
-            let pool = r2d2::Pool::builder()
-                .max_size(size)
-                .build(ConnectionManager::new(parsed))?;
-            connections.push(pool);
-        }
-        Ok(Client {
-            connections,
-            hash_function: default_hash_function,
-        })
-    }
+    /// Connect to memcache via given target Url
+    pub fn connect(target: impl AsRef<str>) -> Result<Self, MemcacheError> {
+        let pool = r2d2::Pool::builder()
+            .max_size(20)
+            .connection_timeout(Duration::from_millis(500))
+            .build(ConnectionManager::new(target.as_ref())?)?;
 
-    pub fn with_pool(pool: Pool<ConnectionManager>) -> Result<Self, MemcacheError> {
-        Ok(Client {
-            connections: vec![pool],
-            hash_function: default_hash_function,
-        })
+        Ok(Self::with_pool(pool))
     }
 
-    pub fn connect<C: Connectable>(target: C) -> Result<Self, MemcacheError> {
-        Self::with_pool_size(target, 1)
+    /// Get pool connection
+    pub fn get_connection(&self) -> Result<PooledConnection<ConnectionManager>, r2d2::Error> {
```

Comment:
Do you own `MemcacheErrror`? I'd consider moving `r2d2::Error` to MemcacheError variants.

Reply:
yep, good point will use MemcacheError